src/Repository/ProgrammeRepository.php

```diff
@@ -77,20 +89,20 @@ public function getPaginated(int $page, int $limit): array
     public function getPaginatedFilteredSorted(
         array $paginate,
         array $filters,
-        string $sort,
-        string $direction
+        ?string $sort,
+        ?string $direction
     ): array {
         $query = $this->entityManager
             ->createQueryBuilder()
             ->select('p')
             ->from('App\Entity\Programme', 'p')
-            ->setFirstResult(($paginate['currentPage'] * $paginate['maxPerPage']) - $paginate['maxPerPage'])
-            ->setMaxResults($paginate['maxPerPage']);
+            ->setFirstResult($paginate['size'] * ($paginate['page'] - 1))
+            ->setMaxResults($paginate['size']);
 
         foreach ($filters as $key => $value) {
-            if ('' != $value) {
-                $query = $query->where("p.$key = :$key");
-                $query->setParameter(':key', $value);
+            if (in_array($key, self::PROGRAMME_FIELDS) && null != $value) {
+                $query->andWhere("p.$key LIKE :value");
```

Comment:
Are you sure you can use the LIKE operator with all those PROGRAMME_FIELDS? Not all of them hold string values, and some don't make sense to be used with LIKE. Also :value remains the same between filters, have you checked there is no overwriting of the parameter value?